backend/tests/unit/app/core/test_logging.py

```
@@ -0,0 +1,128 @@
+            # Should be valid JSON
+            try:
+                log_data = json.loads(log_output)
+                assert "message" in log_data
+                assert log_data["message"] == test_message
+                assert "level" in log_data
+                assert log_data["level"] == "INFO"
+                assert "timestamp" in log_data
+                assert "logger" in log_data
+                assert log_data["logger"] == logger_name
+            except json.JSONDecodeError:
+                assert False, f"Production logs should be JSON formatted. Got: {log_output}"
```

Ruff (0.13.1)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)